pkg/space/space_test.go

```diff
@@ -53,3 +53,30 @@ func TestNewSubSpace(t *testing.T) {
 	)
 	assert.Equal(t, expectedSubSpace, subSpace)
 }
+
+func TestNewSubSubSpace(t *testing.T) {
+	// given
+	srClusterRoles := []string{commoncluster.RoleLabel(commoncluster.Tenant)}
+	sr := spacerequesttest.NewSpaceRequest("jane", "jane-tenant",
+		spacerequesttest.WithTierName("appstudio"),
+		spacerequesttest.WithTargetClusterRoles(srClusterRoles))
+	parentSpace := spacetest.NewSpace(test.HostOperatorNs, "parentSpace")
+	subSpace := NewSubSpace(sr, parentSpace)
+
+	// when
+	subSubSpace := NewSubSpace(sr, subSpace)
+
+	// then
+	expectedSubSubSpace := spacetest.NewSpace(test.HostOperatorNs, SubSpaceName(subSpace, sr),
+		spacetest.WithSpecParentSpace(subSpace.GetName()),
+		spacetest.WithTierName("appstudio"),
+		spacetest.WithSpecTargetClusterRoles([]string{cluster.RoleLabel(cluster.Tenant)}),
+		spacetest.WithLabel(toolchainv1alpha1.SpaceRequestLabelKey, sr.GetName()),
+		spacetest.WithLabel(toolchainv1alpha1.SpaceRequestNamespaceLabelKey, sr.GetNamespace()),
+		spacetest.WithLabel(toolchainv1alpha1.ParentSpaceLabelKey, subSpace.GetName()),
+	)
+	assert.Equal(t, expectedSubSubSpace, subSubSpace)
+
+	// also assert that names don't grow in length as we increase nesting
+	assert.Equal(t, len(subSpace.Name), len(subSubSpace.Name))
```

Comment:
and also that the names are different 
```suggestion
	assert.Equal(t, len(subSpace.Name), len(subSubSpace.Name))
	assert.NotEqual(t, subSpace.Name, subSubSpace.Name)
```

Reply:
Fixed in 35eb34a.